Possible workaround to #11

Jon2G opened the pull request:
Possible workaround to #11
https://github.com/dotnet/maui/issues/19576#issuecomment-2131584486
Also review InitFreakyEffects on MauiProgram sample it fails due to "An item with the same key has already been added. Key: Maui.FreakyEffects.TouchEffects.TouchRoutingEffect'"
![Recording 2024-05-25 231953](https://github.com/FreakyAli/Maui.FreakyEffects/assets/24820069/3d4f620d-42e9-4ad4-81c0-96c82784794b)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added support for .NET 9.0 across multiple platforms, enabling enhanced compatibility.

- **Improvements**
  - Upgraded package dependencies, including `CommunityToolkit.Maui` and `Microsoft.Maui.Controls`, to ensure access to the latest features.
  - Enhanced UI styling in sample views, leading to clearer labels and improved readability.
  - Optimized touch effect handling for consistent color rendering on Android.
  - Introduced additional attributes to animation classes, improving their flexibility with service providers.

- **Bug Fixes**
  - Disabled the problematic initialization responsible for duplicate key exceptions during effect setup.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Jon2G committed:
Possible workaround to bug https://github.com/dotnet/maui/issues/19576#issuecomment-2131584486

Possible workaround to bug https://github.com/dotnet/maui/issues/19576#issuecomment-2131584486

Jon2G commented on Maui.FreakyEffects/Samples/MauiProgram.cs:
```diff
@@ -19,7 +18,8 @@ public static MauiApp CreateMauiApp()
             })
             .ConfigureEffects(effects =>
             {
-                effects.InitFreakyEffects();
```
This should not fail?

FreakyAli replied:
@Jon2G You can take a pull on master now, I have added the fix for this

Jon2G commented on Maui.FreakyEffects/Samples/Samples.csproj:
```diff
@@ -58,9 +58,9 @@
-		<PackageReference Include="Microsoft.Maui.Controls" Version="8.0.21" />
-		<PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="8.0.21" />
-		<PackageReference Include="FreakyControls" Version="0.4.9-pre" />
+		<PackageReference Include="Microsoft.Maui.Controls" Version="8.0.40" />
```
Update project dependencies

Jon2G commented on Maui.FreakyEffects/Maui.FreakyEffects/Shared/Skeleton/SkeletonEffect.cs:
```diff
@@ -187,7 +187,7 @@ private static void RestoreBackgroundColor(View view)
         else if (backgroundColor != default(Color))
         {
-            view.BackgroundColor = GetOriginalBackgroundColor(view);
+            view.BackgroundColor = GetOriginalBackgroundColor(view) ?? Colors.Transparent;
```
If OriginalBackgroundColor is null it will not change so setting to transparent instead seems to do the trick

FreakyAli replied:
Interesting let me take a look at this

FreakyAli commented:
I just checked this, it might fix the issue with the background colour but still won't fix, the text colour part, now the thing is that the text colour can't be set to transparent or to some colour by default as there is no default here if you know what i mean

Jon2G commented:
Yeah i was thinking about that, but text color should never possibly be null
Let me take a look

FreakyAli committed:
Merge branch 'master' into master

Jon2G committed:
Use ClearValue instead of setting color to null

As @PureWeen suggested calling "ClearValue" instead of setting property to null seems to work.
https://github.com/dotnet/maui/issues/22914#issuecomment-2365290936

Jon2G commented:
Hi, can you take a look a the new changes at discussion at https://github.com/dotnet/maui/issues/22914 :)

Jon2G committed:
update samples project target to net9

Jon2G committed:
add net9 target

Jon2G committed:
add support for font icons

Jon2G committed:
support for net9

* add AcceptEmptyServiceProvider to remove warning
* if animation not set use fade animation by default
* support for net9

Jon2G committed:
Code factor brace issue

Jon2G committed:
Update Samples.csproj
